batch/src/main/java/org/openmrs/analytics/JdbcFetchUtil.java

```diff
@@ -145,10 +136,9 @@ public void processElement(@Element KV<String, Iterable<String>> element,
 		}
 	}
 	
-	public Integer fetchMaxId(String tableName) throws SQLException {
+	public Integer fetchMaxId(String tableName) throws SQLException, PropertyVetoException {
 		String tableId = tableName + "_id";
-		Connection con = this.comboPooledDataSource.getConnection();
-		Statement statement = con.createStatement();
+		Statement statement = jdbcConnectionUtil.createStatement();
```

Comment:
Noted and thanks. Happy to fix this!

Reply:
Looking at this, I see that we are indeed closing both the statement and results. Do we need to close the close the connection pool given that it will be used by beam downstream the pipeline?

https://github.com/GoogleCloudPlatform/openmrs-fhir-analytics/blob/c42d81d2c3235e4c76863ef7d8521b56adcf979f/batch/src/main/java/org/openmrs/analytics/JdbcFetchUtil.java#L146-L147

Reply:
Yes, GitHub was hiding the `close()` statements and I got confused. I actually later looked into fixing it but realized they are already there (updated those to try-with-resources pattern). Sorry I should have updated this comment.